### PR TITLE
bug(build): fixed built ESM files importing CJS files from react-styles

### LINKED ISSUE #8871: Bug - [All] - [react core esm packages should explicitely import the .mjs files from react-styles]
**Describe the problem**
In cockpit we are importing files like import "@patternfly/react-core/dist/esm/components/Button/index.js";

I was recently analyzing the bundled output and noticed that the files that are imported from react-styles are not the .mjs ones but the plain CommonJS. This results in the use of 'require' instead of 'import'.

Screenshot from 2023-03-27 23-44-34

**Expected behavior**
I expect that the esm/ directory imports directly from the .mjs files from react-styles.

**Is this issue blocking you?**
No

### PR DESCRIPTION
**Pull request by wise-king-sullyman**
**What**: Closes #8871 

**Additional issues**:

Adds a transformer which updates ESM builds to import the .mjs react-styles css files explicitly, does not change CJS files.

Previous ESM file:
image

New ESM file:
image

### CODE REVIEW

**Review by dlabaj — APPROVED** [review]

File: jest.config.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ module.exports = {
   ],
   roots: ['<rootDir>/packages'],
   transform: {
-    '^.+\\.[jt]sx?$': 'babel-jest'
+    '^.+\\.m?[jt]sx?$': 'babel-jest'
```
Comment (nicolethoen) [inline jest.config.js:13]: what does this check?

Comment (wise-king-sullyman) [reply jest.config.js:13]: Makes it so that the transformer also picks up .mjs files, which are what the react-styles build process outputs for ESM.

**Comment by nicolethoen** [comment]
I intend to pull down that branch and run the build to see what the output looks like - make sure that the transformation doesn’t have unintended consequences by doing a diff against main.

**Comment by nicolethoen** [comment]
I pulled down this branch, rebased it to update it with the latest from main, and then ran a build locally (once on the main branch and once in this updated branch).

I am seeing identical dist directories on the main branch and on this PRs branch. I am not seeing the new mjs import path in the ESM directory like the image in this PR. Is there an extra build step i might be missing?

**Comment by wise-king-sullyman** [comment]
@nicolethoen @dlabaj try now

**Review by mfrances17 — APPROVED** [review]
LGTM, i pulled down your branch and built, and i see the changes

for example, in your branch, for accordion i see this:
`import styles from '@patternfly/react-styles/css/components/Accordion/accordion.mjs';
`

whereas in main it is:
`import styles from '@patternfly/react-styles/css/components/Accordion/accordion';
`

**Review by nicolethoen — APPROVED** [review]
LGTM2 :) I ran it locally and see the transformation now

**Pull request merged**